src/main/java/io/odpf/depot/http/request/BatchRequest.java

```diff
@@ -54,10 +57,9 @@ public List<HttpRequestRecord> createRecords(List<OdpfMessage> messages) {
         }
         if (validBodies.size() != 0) {
             Map<String, String> requestHeaders = headerBuilder.build();
-            URI requestUrl = uriBuilder.build();
-            HttpEntityEnclosingRequestBase request = RequestMethodFactory.create(requestUrl, httpMethod);
-            requestHeaders.forEach(request::addHeader);
-            request.setEntity(RequestUtils.buildStringEntity(validBodies.values()));
+            Map<String, String> queryParam = queryParamBuilder.build();
```

Comment:
if all headerBuilder.build(),queryParamBuilder.build(),and uriBuilder.build(queryParam)
returns the same thing, we should save this and use the same object, They are read only,